forge-gui/res/cardsfolder/upcoming/kogla_and_yidaro.txt

```diff
@@ -0,0 +1,13 @@
+Name:Kogla and Yidaro
+ManaCost:2 R R G G
+Types:Legendary Creature Ape Dinosaur Turtle
+PT:7/7
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigCharm | TriggerDescription$ When CARDNAME enters the battlefield, ABILITY
+SVar:TrigCharm:DB$ Charm | Choices$ DBPump,DBFight
+SVar:DBPump:DB$ Pump | Defined$ Self | KW$ Trample & Haste | SpellDescription$ It gains trample and haste until end of turn.
+SVar:DBFight:DB$ Fight | Defined$ TriggeredCardLKICopy | ValidTgts$ Creature.YouDontCtrl | TgtPrompt$ Choose target creature you don't control | SpellDescription$ CARDNAME fights target creature you don't control.
+A:AB$ Destroy | Cost$ 2 R G Discard<1/CARDNAME> | ValidTgts$ Artifact,Enchantment | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target artifact or enchantment | ActivationZone$ Hand | SubAbility$ DBChangeZone | SpellDescription$ Destroy up to one target artifact or enchantment. Shuffle CARDNAME into your library from your graveyard, then draw a card.
+SVar:DBChangeZone:DB$ ChangeZone | Origin$ Graveyard | Destination$ Library | Shuffle$ True | Defined$ DiscardedCards | SubAbility$ DBDraw
```

Comment:
Is this the best Defined given the Oracle wording?

Maybe `Self` or `CorrectedSelf` ?

Reply:
tool4ever suggested that we replace `CorrectedSelf` with `DiscardedCards`, can you advise @tool4ever ?

Reply:
it's correct :)